File: DDB.HPApi/Models/Character.cs
using System.ComponentModel.DataAnnotations;

namespace DDB.HPApi.Models
{
    public class Character
    {
        [Key]
        public Guid Id { get; set; }

        public string? Name { get; set; }

        public int Level { get; set; }

        public int HitPoints { get; set; }  // Max hit points

        public int TempHitPoints { get; set; }

        public int CurrentHitPoints { get; set; }

        public ICollection<CharacterClass>? Classes { get; set; }

        public Stats? Stats { get; set; }
        public ICollection<Item>? Items { get; set; }

        public ICollection<DefenseType>? Defenses { get; set; }
    }
}

File: ddb-hp-ui/src/app/interfaces/models/character.ts
import Class from "./class"
import Defense from "./defense"
import Item from "./item"
import Stats from "./stats"

export default interface Character {
    id: string
    name: string
    level: number
    hitPoints: number
    tempHitPoints: number
    currentHitPoints: number
    classes: Class[]
    stats: Stats
    items: Item[]
    defenses: Defense[]
}